ibis/expr/operations.py

```diff
@@ -3623,23 +3624,17 @@ class GeoAsText(GeoSpatialUnOp):
 class ElementWiseVectorizedUDF(ValueOp):
     """Node for element wise UDF."""
 
-    func = Arg(callable)
-    func_args = Arg(tuple)
-    input_type = Arg(rlz.shape_like('func_args'))
-    _output_type = Arg(rlz.noop)
-
-    def __init__(self, func, args, input_type, output_type):
-        self.func = func
-        self.func_args = args
-        self.input_type = input_type
-        self._output_type = output_type
+    func = Arg((types.FunctionType, types.LambdaType))
+    func_args = Arg(rlz.list_of(rlz.any, coerce_to=list))
+    input_type = Arg(rlz.list_of(rlz.datatype, coerce_to=list))
+    return_type = Arg(rlz.datatype)
 
     @property
     def inputs(self):
         return self.func_args
 
     def output_type(self):
-        return self._output_type.column_type()
+        return rlz.shape_like(self.func_args, dtype=self.return_type)
```

Comment:
`my_add_function(1.0, 2.0)` returns with a scalar type rather than a column type - presuming that the `func_arg` are correctly converted to ibis literals.

Reply:
what is `my_add_function`?

Reply:
I am ok with this change but also seems a bit unnecessary to handle scalar types for this class.

The original thinking was that the vectorized udf to not handle scalars at all for simplicity (hence the same "vectorized"). If a scalar is passed it, it should probably be converted to a column literal. i.e., "1.0" meaning a vector of all 1.0s

Reply:
Sorry, missed the context. 

I ported a test case from the [pandas backend](https://github.com/ibis-project/ibis/blob/master/ibis/backends/pandas/tests/test_udf.py#L115-L118) to the [datafusion backend](https://github.com/ibis-project/ibis/pull/2918/files#diff-009e15b368654dd7b798e7acef1ac005a55a91a8f815515ae3fadf31f37e90cdR49-R52) and got shape errors. 

Indeed, vectorized should imply that we handle array-like inputs, though the pandas backend tests it for scalar inputs. 
I'm fine with keeping array-like output for vectorized UDFs, but than we may need non-vectorized user defined functions as well?

Reply:
Let's keep it unchanged for now. It seems like a use case that isn't necessary or desirable to support at the moment.

Reply:
I can't really keep it unchanged since we do validate the inputs now.

It would make sense to support both column and array scalar inputs:

```py
class VectorizedUDF(ValueOp):
    func = Arg(rlz.instance_of((FunctionType, LambdaType)))
    func_args = Arg(
        rlz.list_of(rlz.one_of([rlz.column(rlz.any), rlz.scalar(rlz.array)]))
    )
    input_type = Arg(rlz.list_of(rlz.datatype))
    return_type = Arg(rlz.datatype)
```

Though I'm not sure how could we handle the mixture of those input types without explicit transpose of arrays to columns.